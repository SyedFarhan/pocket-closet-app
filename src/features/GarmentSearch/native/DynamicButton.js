import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';

import SearchButton from './SearchButton';
import ClearButton from './ClearButton'

const DynamicButton = ({ onSearch, onClear, isSearched }) => {
  return (
    isSearched ? <ClearButton onClear={onClear} /> : <SearchButton onSearch={onSearch} />
  );
};

// DynamicButton.propTypes = {
//   isSearched: PropTypes.bool.isRequired,
// };

export default DynamicButton;
