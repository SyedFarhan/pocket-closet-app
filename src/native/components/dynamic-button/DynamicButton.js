import React from 'react';
import PropTypes from 'prop-types';

import SearchButton from '../search-button/SearchButton';
import ClearButton from '../clear-button/ClearButton';

const DynamicButton = ({ isSearched, onSearch, onClear }) => {
  return (
    isSearched ? <ClearButton onClear={onClear} /> : <SearchButton onSearch={onSearch} />
  );
};

DynamicButton.propTypes = {
  isSearched: PropTypes.bool.isRequired,
  onSearch: PropTypes.func,
  onClear: PropTypes.func,
};

export default DynamicButton;
