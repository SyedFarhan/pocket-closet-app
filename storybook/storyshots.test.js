import initStoryshots from '@storybook/addon-storyshots';
import { defaultDynamicButtonSpecs, searchedDynamicButtonSpecs } from '../src/features/GarmentSearch/native/DynamicButton.test';

initStoryshots();
