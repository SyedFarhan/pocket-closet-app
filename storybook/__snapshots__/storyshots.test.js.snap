// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots DynamicButton after search 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.25,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#d9534f",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 2,
          "flex": 1,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "width": "100%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
        virtual={undefined}
      >
        Clear
      </Text>
    </View>
  </View>
</View>
`;

exports[`Storyshots DynamicButton default 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#007aff",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "elevation": 2,
      "flex": 1,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
    virtual={undefined}
  >
    Search
  </Text>
</View>
`;

exports[`Storyshots Welcome to Storybook 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 18,
      }
    }
  >
    Welcome to React Native Storybook
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 12,
        "lineHeight": 18,
        "marginBottom": 10,
      }
    }
  >
    This is a UI Component development environment for your React Native app. Here you can display and interact with your UI components as stories. A story is a single state of one or more UI components. You can have as many stories as you want. In other words a story is like a visual test case.
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 12,
        "lineHeight": 18,
        "marginBottom": 10,
      }
    }
  >
    We have added some stories inside the "storybook/stories" directory for examples. Try editing the "storybook/stories/Welcome.js" file to edit this message.
  </Text>
</View>
`;
